### DISCUSSION TITLE
Custom pages / integrations

### DISCUSSION THREAD

**Discussion by Fusseldieb** [open-webui/open-webui#7603] [Ideas]
# Feature Request

**Is your feature request related to a problem? Please describe.**
It would be great to extend OpenWebUI by allowing users to create custom 'pages'/integrations. Currently, there’s no simple way to add personalized features or workflows directly into the interface. It would open a world of possibilities.

**Describe the solution you'd like**
Support for custom integrations, where users can upload a Python script that defines both the menu placement (e.g., in the profile, settings, or the context menu above the profile) and the functionality of the integration itself. Similar to "Functions," but designed to expose a page with custom functionality. This would enable features such as a custom usage tracker, specialized dashboards, or embedded tools.

**Describe alternatives you've considered**
- Building standalone tools that interact with OpenWebUI via API, but this lacks integration.
- Modifying the core codebase, which isn’t user-friendly or sustainable.

**Additional context**
For example, users could create a dashboard for managing billing or whatsoever, which could be especially valuable if the instance is shared. 

This would make OpenWebUI much more flexible and personalized.

**Comment by silenceroom** [comment]
This sounds like a great idea. 

Further thoughts: the workspace currently supports prompts/tools/KB. If a new "custome Page" module can be added which allow approved users to create custom pages within the style, this would really turn OWUI into a real "AI Portal" concept platform. 

A few example: 
1. Many systems like Dify/Voiceflow etc supports to publish workflow into a single-page app, this can be integreated via a 'iframe' wihtin OWUI for direct access. 
2. Build custom pages with external APIs for complicated workflows. Currently OWUI is still limitted to simple chat-based response, with custom pages (with API interaction), it can be easily turn into a real productive tool.
